python/tank/authentication/sso_saml2/core/username_password_dialog.py

```diff
@@ -0,0 +1,155 @@
+
+        def __init__(self):
+            super(UsernamePasswordDialog, self).__init__()
+
+            # For now we fix the GUI size.
+            self.setWindowTitle("Authentication Required")
+            self.setMinimumWidth(420)
+            self.setMinimumHeight(120)
+
+            # set up the layout
+            form_grid_layout = QtGui.QGridLayout(self)
+
+            # initialize the username combo box so that it is editable
+            self._edit_username = QtGui.QLineEdit(self)
+
+            # initialize the password field so that it does not echo characters
+            self._edit_password = QtGui.QLineEdit(self)
+            self._edit_password.setEchoMode(QtGui.QLineEdit.Password)
+
+            # initialize the labels
+            label_username = QtGui.QLabel(self)
+            label_password = QtGui.QLabel(self)
+            label_username.setText("Username:")
+            label_password.setText("Password:")
+
+            # initialize buttons
+            buttons = QtGui.QDialogButtonBox(self)
+            buttons.addButton(QtGui.QDialogButtonBox.Ok)
+            buttons.addButton(QtGui.QDialogButtonBox.Cancel)
+            buttons.button(QtGui.QDialogButtonBox.Ok).setText("Login")
+            buttons.button(QtGui.QDialogButtonBox.Cancel).setText("Cancel")
+
+            # place components into the dialog
+            form_grid_layout.addWidget(label_username, 0, 0)
+            form_grid_layout.addWidget(self._edit_username, 0, 1)
+            form_grid_layout.addWidget(label_password, 1, 0)
+            form_grid_layout.addWidget(self._edit_password, 1, 1)
+            form_grid_layout.setRowStretch(2, 1)
+            form_grid_layout.addWidget(buttons, 3, 1)
+
+            self.setLayout(form_grid_layout)
+
+            buttons.button(QtGui.QDialogButtonBox.Ok).clicked.connect(self._on_enter_credentials)
+            buttons.button(QtGui.QDialogButtonBox.Cancel).clicked.connect(self.close)
+
+        @property
+        def username(self):
+            """Getter for username."""
+            return self._edit_username.text()
+
+        @username.setter
+        def username(self, username):
+            """Setter for username."""
+            self._edit_username.setText(username)
+
+        @property
+        def password(self):
+            """Getter for password."""
+            return self._edit_password.text()
+
+        @password.setter
+        def password(self, password):
+            """Setter for password."""
+            self._edit_password.setText(password)
+
+        def _on_enter_credentials(self):
+            """Callback when clicking Ok."""
+            if self._edit_username.text() == "":
+                self._edit_username.setFocus()
+                return
+
+            if self._edit_password.text() == "":
+                self._edit_password.setFocus()
+                return
+
+            self.accept()
+except ImportError:
+    class UsernamePasswordDialog(object):
```

Comment:
What's the situation that exists where our Qt abstraction isn't available and you would need this faux dialog object? I'd prefer to assume we can import Qt in this module and provide a single, real implementation of the dialog, and then handle the error case in the higher level calling code.

Reply:
I got confused by the results of the AppVeyor check. You are correct, the mock version is not useful. I have removed it.